modules/broker-amqp/src/main/java/io/ballerina/messaging/broker/amqp/codec/InMemoryMessageAggregator.java

```diff
@@ -138,4 +148,25 @@ public Message popMessage() {
     public void setTransaction(BrokerTransaction transaction) {
         this.transaction = transaction;
     }
+
+    public void publishMessage(ChannelHandlerContext ctx, AmqpChannel channel, int channelID) {
+        Message publishMessage = popMessage();
+        // flow manager should always be executed through the event loop
+        ctx.fireChannelRead((BlockingTask) () -> {
```

Comment:
ctx.fireChannel code segment should go into header frame IMO

Reply:
I will correct it ayye

Reply:
There seems to be a test failure calling both popMessage() inside ctx.fireChannel code. Is it better to add popMessage() and then publish() inside ctx.fireChannel code in header frame and content frame?